integration-tests/tests/snapd_test.go

```diff
@@ -223,26 +216,42 @@ func doMethodNotAllowed(c *check.C, resource, verb string) {
 		resource,
 		verb,
 		apiInteraction{
-			responsePattern: `{"result":{},"status":"Unauthorized","status_code":401,"type":"error"}`})
+			responsePattern: `{"result":{},"status":"Method Not Allowed","status_code":405,"type":"error"}`})
 }
 
 // this is the function which makes requests to the api
-func genericRequest(resource, verb string, payload io.Reader) (body []byte, err error) {
-	if file, ok := payload.(*os.File); ok {
-		defer file.Close()
+func genericRequest(options *requestOptions) (body []byte, err error) {
```

Comment:
The convention for Go documentation is something along those lines:

    // genericRequest makes a request to the API according to the provided options.

I'd probably also name it `doRequest` or `makeRequest` instead, both to have a verb in there and because "generic" doesn't seem to add value in this case.